node/tests/SharedTests.ts

```diff
@@ -5743,6 +5743,156 @@ export function runBaseTests<Context>(config: {
         config.timeout,
     );
 
+    it.each([ProtocolVersion.RESP2, ProtocolVersion.RESP3])(
+        "dump and restore test_%p",
+        async (protocol) => {
+            await runTest(async (client: BaseClient) => {
+                const key1 = "{key}-1" + uuidv4();
+                const key2 = "{key}-2" + uuidv4();
+                const key3 = "{key}-3" + uuidv4();
+                const nonExistingkey = "{nonExistingkey}-" + uuidv4();
+                const value = "orange";
+                const valueEncode = Buffer.from(value);
+
+                expect(await client.set(key1, value)).toEqual("OK");
+
+                // Dump non-existing key
+                expect(await client.dump(nonExistingkey)).toBeNull();
+
+                // Dump existing key
+                const data = await client.dump(key1);
+                expect(data).not.toBeNull();
+
+                if (data != null) {
```

Comment:
You can remove this - if `data` is null, you'll have an error on the previous line.

Reply:
I add this check to avoid the error raised by VS Code.
```
Argument of type 'Buffer | null' is not assignable to parameter of type 'Buffer'.
  Type 'null' is not assignable to type 'Uint8Array'.ts(2345)
```

Reply:
So you can update line 5763:
```
const data = await client.dump(key1) as Buffer;
```
or
```
const data = (await client.dump(key1)) as Buffer;
```

... or keep it as is